### PR TITLE
k8s.io/component-base/metrics/testutil/:improve code

### PR DESCRIPTION
**Pull request by tanjunchen**
**What type of PR is this?**
 
/kind cleanup
 

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
None
```

### CODE REVIEW

File: staging/src/k8s.io/component-base/metrics/testutil/metrics.go [hunk 1]
```diff
@@ -207,22 +207,22 @@ func GetHistogramFromGatherer(gatherer metrics.Gatherer, metricName string) (His
...
-		if mFamily.Name != nil && *mFamily.Name == metricName {
+		if mFamily.GetName() == metricName {
 			metricFamily = mFamily
 			break
 		}
 	}
 
 	if metricFamily == nil {
-		return Histogram{}, fmt.Errorf("Metric %q not found", metricName)
+		return Histogram{}, fmt.Errorf("metric %q not found", metricName)
```
Comment (tanjunchen) [inline staging/src/k8s.io/component-base/metrics/testutil/metrics.go:217]: the staticcheck in my idea

File: staging/src/k8s.io/component-base/metrics/testutil/metrics.go [hunk 2]
```diff
@@ -277,12 +277,12 @@ func bucketQuantile(q float64, buckets []bucket) float64 {
 // Quantile computes q-th quantile of a cumulative histogram.
 // It's expected the histogram is valid (by calling Validate)
 func (hist *Histogram) Quantile(q float64) float64 {
-	buckets := []bucket{}
+	var buckets []bucket
```
Comment (tanjunchen) [inline staging/src/k8s.io/component-base/metrics/testutil/metrics.go:280]: Empty slice declaration via literal

**Review by logicalhan — APPROVED** [review]
/lgtm
/approve

**Pull request merged**